app/models/grda_warehouse/hud/client.rb

```diff
@@ -2041,7 +2041,7 @@ def veteran?
     end
 
     def ever_veteran?
-      source_clients.map(&:veteran?).include?(true)
+      veteran_override? || source_clients.map(&:veteran?).include?(true)
     end
 
     def adjust_veteran_status
```

Comment:
Merge ClientCleanup#choose_best_veteran_status with various methods on Client.  Probably this will end up as a class method on client that accepts dest_attr and source_client_attrs.